folium/raster_layers.py

```diff
@@ -104,14 +106,26 @@ def __init__(
         subdomains: str = "abc",
         tms: bool = False,
         opacity: float = 1,
-        **kwargs
+        **kwargs,
     ):
-        # check for xyzservices.TileProvider without importing it
-        if isinstance(tiles, dict):
+        if isinstance(tiles, str):
+            if tiles.lower() == "openstreetmap":
+                tiles = "OpenStreetMap Mapnik"
+                if name is None:
+                    name = "openstreetmap"
+            try:
+                tiles = xyzservices.providers.query_name(tiles)
+            except ValueError:
+                # no match, likely a custom URL
+                pass
+
+        if isinstance(tiles, xyzservices.TileProvider):
```

Comment:
question, any particular reason to not use `elif` here?

Reply:
A string is used as an input to `xyzservices.providers.query_name` on l. 117 which produces TileProvider that needs to be parsed here. `elif` would not work.

Reply:
That makes sense, thanks!